adoc/chapters/programming_interface.adoc

```diff
@@ -17640,40 +17635,500 @@ member function template.
+!====
+_Returns:_ A reference to the element of the underlying [code]#vec# object that
+corresponds to the position [code]#index# of the swizzle operation.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+template</*unspecified*/>
+__swizzled_vec__& operator=(const __swizzled_vec__</*unspecified*/>& rhs)
+----
+!====
+_Constraints:_ Available only when the element data type of [code]#rhs# is the
+same as [code]#DataT# and when the number of elements in the [code]#rhs# view
+is equal to [code]#NumElements#.
+
+_Effects:_ Assigns elements from the [code]#rhs# [code]#+__swizzled_vec__+#
+view to elements of "this" [code]#+__swizzled_vec__+# view.
+The value from the [code]#rhs# corresponding to the first element of the
+[code]#rhs# swizzle operation is assigned to the element of the underlying
+[code]#vec# object that corresponds to the first element of the "this" swizzle
+operation, etc.
+
+The order in which the elements are assigned is unspecified.
+Therefore, if the "this" view has any repeated elements in the swizzle, those
+elements may receive unspecified values.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+__swizzled_vec__& operator=(const DataT& rhs)
+----
+!====
+_Effects:_ Assigns the value [code]#rhs# to those elements of the underlying
+[code]#vec# object that have corresponding elements in the
+[code]#+__swizzled_vec__+# view.
+Elements in the underlying [code]#vec# object that do not have elements in the
+[code]#+__swizzled_vec__+# view are not assigned.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+__swizzled_vec__& operator=(const vec<DataT, NumElements>& rhs)
+----
+!====
+_Effects:_ Assigns elements from [code]#rhs# to elements of the [code]#vec#
+object that underlies this [code]#+__swizzled_vec__+# view.
+The first element of [code]#rhs# is assigned to the element of the underlying
+[code]#vec# object that corresponds to the first element of the swizzle
+operation, etc.
+
+The order in which the elements are assigned is unspecified.
+Therefore, if the [code]#+__swizzled_vec__+# view has any repeated elements in
+the swizzle, those elements may receive unspecified values.
+
+_Returns:_ A reference to the "this" [code]#+__swizzled_vec__+# view.
+|====
+
+[[table.functions.swizzled-vec]]
+.Hidden friend functions of the [code]#+__swizzled_vec__+# class template
+[separator="@"]
+|====
+a@
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+template</*unspecified*/>                                                                 (1)
+friend vec<DataT, NumElements> operatorOP(const __swizzled_vec__& lhs,
+                                          const __swizzled_vec__</*unspecified*/>& rhs)
+
+friend vec<DataT, NumElements> operatorOP(const vec<DataT, NumElements>& lhs,             (2)
+                                          const __swizzled_vec__& rhs);
```

Comment:
Some of the `__swizzled_vec__` types are missing the `</*unspecified*/>` part, is that intentional?

Reply:
That's a very good point, this is definitely the right way to do this, I missed that having an unspecified template parameter pack in multiple places could allow for those to have different template parameters which is not what we want.

Reply:
Thanks for adding the clarification.